src/Security/PasswordEncryptorPBKDF2.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace SilverStripe\Security;
+
+use RuntimeException;
+
+/**
+ * Provides Password-Based Key Derivation Function hashing for passwords, using the provided algorithm (default
+ * is SHA512), which is NZISM compliant under version 3.2 section 17.2.
+ */
+class PasswordEncryptorPBKDF2 extends PasswordEncryptor
+{
+    private string $algorithm = 'sha512';
+
+    /**
+     * The number of internal iterations for hash_pbkdf2() to perform for the derivation. Please note that if you
+     * change this from the default value you will break existing hashes stored in the database, so these would
+     * need to be regenerated.
+     */
+    private int $iterations = 30000;
+
+    /**
+     * @throws RuntimeException If the provided algorithm is not available in the current environment
+     */
+    public function __construct(string $algorithm, ?int $iterations = null)
+    {
+        if (!in_array($algorithm, hash_hmac_algos())) {
+            throw new RuntimeException(
+                sprintf('Hash algorithm "%s" not found in hash_hmac_algos()', $algorithm)
+            );
+        }
```

Comment:
As per https://www.php.net/manual/en/function.hash-pbkdf2.php

Note this was really the only thing that was shared with `PasswordEncryptor_PHPHash` which is why that's not the superclass for this implementation.